packages/clerk-js/src/utils/componentGuards.ts

```diff
@@ -7,7 +7,7 @@ export type ComponentGuard = (
 ) => boolean;
 
 export const sessionExistsAndSingleSessionModeEnabled: ComponentGuard = (clerk, environment) => {
-  return !!(clerk.session && environment?.authConfig.singleSessionMode);
+  return !!(clerk.isSignedIn && !clerk.session?.hasTask && environment?.authConfig.singleSessionMode);
```

Comment:
We should also add another redirect guard such as: `hasPendingTasksAndSingleSessionModeEnabled` in order to redirect to the task path.

Reply:
Just realized that this is a problem, if the user has a session with a pending task, and tries to go back to sign-up, it'll error out for single session mode.

The right approach here is to navigate to the task when sign-in mounts